src/jukebox/player/PlayerMPD.py

```diff
@@ -32,22 +34,72 @@ def __init__(self, mpd_host, music_player_status, volume_control=None):
-        try:
-            ret = mpd_cmd(params)
-        except ConnectionError:
-            logger.error("MPD Connection Error, retry")
-            self.conncet()
-            ret = mpd_cmd(params)
-        except Exception as e:
-            logger.error(f"{e}")
-        return ret
+    def mpd_retry_with_mutex(self, mpd_cmd, params=None):
+        '''
+        This method adds thread saftey for acceses to mpd via a mutex lock,
+        it shall be used for each access to mpd to ensure thread safety
+        In case of a communication error the connection will be reestablished and the pending command will be repeated 2 times
+        '''
+        retry = 2
+        with self.mpd_mutex:
+            while retry:
+                try:
+                    if params is None:
+                        ret = mpd_cmd()
+                    else:
+                        ret = mpd_cmd(params)
+                    break
+                except ConnectionError:     # TODO: this is not working properly yet, we are alwas anding up in the Exception!
+                    logger.info(f"MPD Connection Error, retry {retry}")
+                    self.connect()
+                    retry -= 1
+                except Exception as e:
+                    if retry:
+                        retry -= 1
+                        self.connect()      # TODO: Workaround, since the above ConnectionError is not properly caught
+                        logger.info(f"MPD Error, retry {retry}")
+                        logger.info(f"{e.__class__}")
+                        logger.info(f"{e}")
+                    else:
+                        logger.error(f"{e}")
+                        ret = {}
+                        break
+            return ret
+
+    def _mpd_status_poll(self):
+        '''
+        this method polls the status from mpd and stores the important inforamtion in the music_player_status,
+        it will repeat itself in the intervall specified by self.mpd_status_poll_interval
+        '''
+        self.mpd_status.update(self.mpd_retry_with_mutex(self.mpd_client.status))
+
+        # get song name just if the song has changed
+        if self.mpd_status.get('song') != self.old_song:
+            self.mpd_status.update(self.mpd_retry_with_mutex(self.mpd_client.currentsong))
+            self.old_song = self.mpd_status['song']
+
+        self.mpd_status['volume'] = self.volume_control.volume
+
+        if self.mpd_status.get('elapsed') is not None:
+            self.current_folder_status["ELAPSED"] = self.mpd_status['elapsed']
+            self.music_player_status['player_status']["CURRENTSONGPOS"] = self.mpd_status['song']
+            self.music_player_status['player_status']["CURRENTFILENAME"] = self.mpd_status['file']
+
+        # the repetation is intentionally at the end, to avoid overruns in case of delays caused by communication
+        self.status_thread = threading.Timer(self.mpd_status_poll_interval, self._mpd_status_poll).start()
```

Comment:
This starts a new thread every poll_intervall ms. I have an idea how we can extend the daemon <-> module interface, so we can make this a true background task. But later ...

Reply:
> This starts a new thread every poll_intervall ms. 
Yes, same concern for me, but seems to be the way it goes. 
Other way would be starting a thread with endless loop and sleeping 250ms each iteration